File: src/main/java/dev/slimevr/config/AutoBoneConfig.java
package dev.slimevr.config;

public class AutoBoneConfig {

	public int cursorIncrement = 2;
	public int minDataDistance = 1;
	public int maxDataDistance = 1;

	public int numEpochs = 100;

	public int printEveryNumEpochs = 25;

	public float initialAdjustRate = 10f;
	public float adjustRateMultiplier = 0.995f;

	public float slideErrorFactor = 0.0f;
	public float offsetSlideErrorFactor = 2.0f;
	public float footHeightOffsetErrorFactor = 0.0f;
	public float bodyProportionErrorFactor = 0.825f;
	public float heightErrorFactor = 0.0f;
	public float positionErrorFactor = 0.0f;
	public float positionOffsetErrorFactor = 0.0f;

	public boolean calcInitError = false;
	public float targetHeight = -1;

	public boolean randomizeFrameOrder = true;
	public boolean scaleEachStep = true;

	public int sampleCount = 1000;
	public long sampleRateMs = 20;

	public boolean saveRecordings = false;

	public AutoBoneConfig() {
	}
}

File: server/src/main/java/dev/slimevr/config/AutoBoneConfig.kt
package dev.slimevr.config

class AutoBoneConfig {
	var cursorIncrement = 2
	var minDataDistance = 1
	var maxDataDistance = 1
	var numEpochs = 100
	var printEveryNumEpochs = 25
	var initialAdjustRate = 10.0f
	var adjustRateDecay = 1.0f
	var slideErrorFactor = 0.0f
	var offsetSlideErrorFactor = 2.0f
	var footHeightOffsetErrorFactor = 0.0f
	var bodyProportionErrorFactor = 0.825f
	var heightErrorFactor = 0.0f
	var positionErrorFactor = 0.0f
	var positionOffsetErrorFactor = 0.0f
	var calcInitError = false
	var targetHeight = -1f
	var randomizeFrameOrder = true
	var scaleEachStep = true
	var sampleCount = 1000
	var sampleRateMs: Long = 20
	var saveRecordings = false
}
